File: solutions/python/difference-of-squares/1/difference_of_squares.py
def square_of_sum(number):
    result=0
    for i in range(1,number+1):
        result=result+i
    result=result*result
    return result

def sum_of_squares(number):
    result=0
    for i in range(1,number+1):
        result=result+(i*i)
    return result

def difference_of_squares(number):
    return square_of_sum(number) - sum_of_squares(number)

File: solutions/r/difference-of-squares/1/difference-of-squares.R
# this is a stub function that takes a natural_number
# and should return the difference-of-squares as described
# in the README.md
sum_of_squares <- function (natural_number) {
  if (natural_number > 0) {
    result <- (natural_number ^ 2) + sum_of_squares(natural_number - 1) 
  }
  else {
    result <- 0
  }
  return(result)
}

sum_of_n <- function (natural_number) {
  if (natural_number > 0) {
    result <- natural_number + sum_of_n(natural_number - 1) 
  }
  else {
    result <- 0
  }
  return(result)
}

difference_of_squares <- function(natural_number) {
  return((sum_of_n(natural_number))^2 - sum_of_squares(natural_number))
}
